File: app/src/main/java/com/practice/forecast/ui/splash/SplashViewModel.java
package com.practice.forecast.ui.splash;

import com.practice.weathermodel.logger.ILog;

import java.util.concurrent.TimeUnit;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;
import io.reactivex.Observable;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.schedulers.Schedulers;

public class SplashViewModel extends ViewModel implements SplashContract.BaseSplashViewModel {
    private final MutableLiveData<Boolean> isReady = new MutableLiveData<>();
    private final ILog logger;
    private final CompositeDisposable compositeDisposable = new CompositeDisposable();

    public SplashViewModel(ILog logger) {
        this.logger = logger;
    }

    @Override
    public LiveData<Boolean> getIsReady() {
        return isReady;
    }

    @Override
    public void startTimer(){
        compositeDisposable.add(Observable.timer(1, TimeUnit.SECONDS)
        .subscribeOn(Schedulers.io())
        .observeOn(AndroidSchedulers.mainThread())
        .subscribe(s -> isReady.setValue(true),
                throwable -> logger.log("SplashViewModel startTimer() error" + throwable.getMessage())));
    }

    @Override
    protected void onCleared() {
        super.onCleared();
        compositeDisposable.clear();
    }
}

File: app/src/main/java/com/practice/forecast/ui/splash/SplashViewModel.kt
package com.practice.forecast.ui.splash

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.practice.forecast.ui.splash.SplashContract.BaseSplashViewModel
import com.practice.weathermodel.logger.ILog
import io.reactivex.Observable
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.CompositeDisposable
import io.reactivex.schedulers.Schedulers
import java.util.concurrent.TimeUnit

class SplashViewModel(private val logger: ILog) : ViewModel(), BaseSplashViewModel {
    private val isReady = MutableLiveData<Boolean>()
    private val compositeDisposable = CompositeDisposable()

    override fun getIsReady(): LiveData<Boolean> {
        return isReady;
    }

    override fun startTimer() {
        isReady.setValue(false)
        compositeDisposable.add(Observable.timer(1, TimeUnit.SECONDS)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe({ _: Long? ->
                    logger.log("SplashViewModel startTimer() success")
                    isReady.setValue(true) }
                ) { throwable: Throwable -> logger.log("SplashViewModel startTimer() error: " + throwable.message) })
    }

    override fun onCleared() {
        super.onCleared()
        compositeDisposable.clear()
    }

}